core-addon/Core.js

```diff
@@ -493,12 +496,40 @@ module.exports = class Core {
       await browser.management.getAll().then(addons =>
         addons.filter(a => a.type == "extension")
           .map(a => a.id));
+
+    // Attempt to resume any paused studies, or pause any running
+    // studies, as appropriate.
+    await this._sendRunState(studies, installedAddonsIds);
+
     return studies.map(s => {
       s.studyInstalled = installedAddonsIds.includes(s.addonId);
       return s;
     });
   }
 
+  /**
+   * Send run state (paused, running) message to study add-on(s).
+   *
+   * @param {Array} studies - list of available studies
+   * @param {Array} installedAddonsIds - list of installed study add-on IDs
+   */
+
+  async _sendRunState(studies, installedAddonsIds) {
```

Comment:
Are you planning on adding an integration test for this? I'm ok with merging this without an integration test as long as we do in depth manual testng :)

Reply:
Yes, honestly I expect the integration test to be fairly simple, but I want to rebase on top of the improvements you made there recently before I start.